src/System.Management.Automation/engine/Subsystem/FeedbackSubsystem/IFeedbackProvider.cs

```diff
@@ -72,31 +156,31 @@ internal GeneralCommandErrorFeedback()
 
                 if (command is not null)
                 {
-                    return StringUtil.Format(
-                        SuggestionStrings.Suggestion_CommandExistsInCurrentDirectory,
-                        target,
-                        localTarget);
+                    return new FeedbackItem(
+                        StringUtil.Format(SuggestionStrings.Suggestion_CommandExistsInCurrentDirectory, target),
+                        new List<string> { localTarget });
                 }
 
                 // Check fuzzy matching command names.
                 if (ExperimentalFeature.IsEnabled("PSCommandNotFoundSuggestion"))
                 {
-                    _fuzzySb ??= ScriptBlock.CreateDelayParsedScriptBlock(@$"
-                        param([string] $target)
-                        $cmdNames = Get-Command $target -UseFuzzyMatching -FuzzyMinimumDistance 1 | Select-Object -First 5 -Unique -ExpandProperty Name
-                        if ($cmdNames) {{
-                            [string]::Join(', ', $cmdNames)
-                        }}
-                    ", isProductCode: true);
-
-                    _args[0] = target;
-                    var result = _fuzzySb.InvokeReturnAsIs(_args);
-
-                    if (result is not null && result != AutomationNull.Value)
+                    var pwsh = PowerShell.Create(RunspaceMode.CurrentRunspace);
+                    var results = pwsh.AddCommand("Get-Command")
+                            .AddParameter("UseFuzzyMatching")
+                            .AddParameter("FuzzyMinimumDistance", 1)
+                            .AddParameter("Name", target)
+                        .AddCommand("Select-Object")
+                            .AddParameter("First", 5)
+                            .AddParameter("Unique")
+                            .AddParameter("ExpandProperty", "Name")
+                        .Invoke<string>();
+
+                    if (results.Count > 0)
                     {
-                        return StringUtil.Format(
+                        return new FeedbackItem(
                             SuggestionStrings.Suggestion_CommandNotFound,
-                            result.ToString());
+                            new List<string>(results),
```

Comment:
It's not read only. The `Next` property is settable, allow a `FeedbackProvider` implementation to return more than 1 feedback items from a call to `GetFeedback`.

Reply:
If add new item as `head` the item can be readonly. 
RecommendedActions could be array, not List.

If think about avoiding breaking change in future the FeedbackItem could be XML. In the case both client and server could be more flexible.

Reply:
> If add new item as head the item can be readonly.

I'm not sure I understand this.

> RecommendedActions could be array, not List.

There is no need to be an array. Even if it's array type, a developer will almost always start with a list and then convert it to an array. That's an unnecessary waste.

> If think about avoiding breaking change in future the FeedbackItem could be XML.

Not sure I understand your concern, but definitely not XML :)

Reply:
> Not sure I understand your concern, but definitely not gonna use XML :)

There was already a breaking change in past and we got many reports (users did not realize that they needed to update PSRL). So I was thinking that there could be version control and a more universal interface to avoid breaking changes in future. It's hardly possible if it's `class`.

As for readonly, it's just my outside view - if there is no intention to change this data on the client, it should be readonly/immutable. So I'm somewhat surprised that it's not. But certainly I don't know how it works in-depth and how it's more convenient for you.

Reply:
We try to avoid breaking changes as much as possible, but that's not a reason to not use a class.
Also, the return type is just one aspect of the interface contract, and changes in parameters would be more likely to happen -- the issue in PSRL you mentioned was because a change in the parameters of the interface method.

Breaking change is inevitable to happen, and we just have to deal with it when it cannot be avoided. Your idea about having PowerShell to show the breaking changes at the first startup is interesting. You should open an issue to continue the discussion on that.

> As for readonly, it's just my outside view 

Thanks! It makes sense to think about making a data structure read-only when it can be or should be.
I thought about making it read-only too, but that comes with extra overhead without real benefit in this scenario:
- developer of a `FeedbackProvider` needs to create `ReadOnlyCollection` for the recommended actions, which is extra allocation and copy per every call to `GetFeedback`.
- the return value will be directly handed to the host, who will do the rendering, and it's a one-way flow. There is no point for the host to change the value returned from `GetFeedback`, because it already can render whatever it wants ...

So, I decided to not pursue a read-only class here.

Reply:
Thanks for clarify!

> Your idea about having PowerShell to show the breaking changes at the first startup is interesting. You should open an issue to continue the discussion on that.

Where was that suggestion? I lost the context. :-) Or was it about experimental features?

Reply:
It's here: https://github.com/PowerShell/PowerShell/issues/14862#issuecomment-1425244618